gh-95994: Clarify escaped newlines. (#96066)
python/cpython: Doc/reference/lexical_analysis.rst

+-----------------+---------------------------------+-------+
| Escape Sequence | Meaning                         | Notes |
+=================+=================================+=======+

> | ``\`` <LF>      | Backslash and newline ignored   | \(1)  |

Reviewer: ```suggestion
| ``\``\ :kbd:`LF`| Backslash and newline ignored   | \(1)  |
```

* the `LF` must be directly after the backtick in actual code, so escape the space
* Use `:kbd:` role that renders similarly to a special character displayed code in most editors, and avoid `<>` which could be confused for a literal

You might need to revise this to add one more space to the table alignment, not sure

Author: I considered using `:kbd:`, but eventually decided against. The <LF> without markup stands out as separate from the `\` literal, and the <> IMHO help indicate that it's a single entity (that correspond to a line feed) and not a string of two characters.  LF is also used in the same table without special markup (and without <>, but in a context where it's not something you type).

For comparison, this is my original version, where the `<LF>` is easily recognizable as not being part of the literal:
![image](https://user-images.githubusercontent.com/25624924/185800397-fbd5afca-a858-4f7c-a0ab-97dbe004cecf.png)

Your suggestion without space almost looks like `\LF` (at least with the current theme):
![image](https://user-images.githubusercontent.com/25624924/185800264-2cf10354-1735-44d3-9496-868523f7cb08.png)

With the space and `:kbd:` is a bit better:
![image](https://user-images.githubusercontent.com/25624924/185800323-6eddabc3-d8d9-482c-bf82-c4fea503ed69.png)

Reviewer: Sorry for the delay—thought I replied to this before.

Ah, I see—on our theme, we render `:kbd:` as something that looks more distinctly like it does on GitHub, i.e.

`\` <kbd>LF</kbd>

This is similar to how control characters are rendered in many/most editors, making it easier for users to infer the intended meaning, uses more appropriate semantic markup, and avoids ambiguous characters that could be interpreted as literals.

However, I inexplicably neglected to check how `:kbd:` renders in the Python docs theme, which of course plays a major role in how this interpreted. As such, it is clear why you made the decision you did—as you say, `:kbd:` makes it much less clear, not more, that `LF` is not part of the literal. I should have checked here, like you did—my mistake, sorry.

So, I agree with your conclusions on the options presented. However, I'm still somewhat unsatisfied with `\` <LF>—it still has a space that could be interpreted as part of a literal, the `<`/`>` can be potentially ambiguous, and it isn't totally obvious to all users what it is intended to actually be. This perhaps gets at a more meaningful concern—`LF` can be quite cryptic to many users who aren't familiar with the ASCII control characters, and furthermore isn't even really accurate—the newline character could just as well be `CRLF` on Windows platforms, and perhaps even others as well.

How about as a potentially better solution:

`\`\<newline\>

This:

* Looks much less like a literal, avoiding any confusion and the need for an extra space
* Is much more clear to a user reading it what it means, without having to be well versed in ASCII control codes
* Is more accurate, since what is escaped is any interpreter-recognized newline character, not `LF` specifically

Reviewer: ```suggestion
| ``\``\ <newline>| Backslash and newline ignored   | \(1)  |
```

Somehow forgot to make this as an actual suggestion, sorry

Author: ![image](https://user-images.githubusercontent.com/25624924/186970636-106da0b7-8a03-4f20-a585-3d4244fccc7c.png)
This looks good enough.

| ``\xhh``        | Character with hex value *hh*   | (3,4) |
+-----------------+---------------------------------+-------+

Escape sequences only recognized in string literals are:

+-----------------+---------------------------------+-------+
| Escape Sequence | Meaning                         | Notes |
+=================+=================================+=======+
| ``\N{name}``    | Character named *name* in the   | \(5)  |
|                 | Unicode database                |       |
+-----------------+---------------------------------+-------+
| ``\uxxxx``      | Character with 16-bit hex value | \(6)  |
|                 | *xxxx*                          |       |
+-----------------+---------------------------------+-------+
| ``\Uxxxxxxxx``  | Character with 32-bit hex value | \(7)  |
|                 | *xxxxxxxx*                      |       |
+-----------------+---------------------------------+-------+

Notes:

(1)
   A backslash can be added at the end of a line to ignore the newline::

      >>> 'This string will not include \
      ... backslashes or newline characters.'
      'This string will not include backslashes or newline characters.'

>    The same result can be achieved by using parentheses and :ref:`string
>    literals concatenation <string-concatenation>` or :ref:`triple-quoted
>    strings <strings>`.

Reviewer: ```suggestion
   The same result can be achieved using :ref:`triple-quoted strings <strings>`,
   or parentheses and :ref:`string literal concatenation <string-concatenation>`.
```

* Avoid ambiguous syntax (it currently can be equally read as "The same result can be achieved by using parenthesis and (string literal concatenation or triple-quoted strings.")
* Use correct concept name/link title (string _literal_, not _literals_, concatenation)
